src/components/CreateProvider.svelte

```diff
@@ -0,0 +1,90 @@
+<script lang="ts">
+    import {dotApi, storeCurrentAction, transactionSigningAddress} from "$lib/stores";
+    import type {DotApi} from '$lib/storeTypes';
+    import {defaultDotApi} from "$lib/storeTypes";
+    import type {ApiPromise} from "@polkadot/api";
+    import {isLocalhost} from "$lib/utils";
+    import {submitCreateProvider} from "$lib/connections";
+    import TransactionStatus from "$components/TransactionStatus.svelte";
+    import {isFunction} from "@polkadot/util";
+    import {onMount} from "svelte";
+
+    let newProviderName = '';
+    let localDotApi: DotApi = defaultDotApi;
+    let web3FromSource;
+    let web3Enable;
+    let showTransactionStatus = false;
+    export let txnStatuses: Array<string> = [];
+    export let validAccounts = {}
+    export let signingAddress = '';
+    export let cancelAction;
+
+    onMount(async () => {
+        const extension = await import('@polkadot/extension-dapp');
+        web3FromSource = extension.web3FromSource;
+        web3Enable = extension.web3Enable;
+    });
+
+    let localSigningAddress = '0xabcdefbeef';
+    transactionSigningAddress.subscribe(addr => localSigningAddress = addr)
+    dotApi.subscribe((api: DotApi) => localDotApi = api)
+
+    const doCreateProvider = async (_evt: Event) => {
+        if (newProviderName === '') {
+            alert("please enter a Provider Name");
+            return;
+        }
+        if (!localDotApi) {
+            alert('please reconnect to an endpoint');
+            return;
+        }
+        clearTxnStatuses();
+        let endpointURI: string = localDotApi.selectedEndpoint || '';
+        let signingKeys = validAccounts[signingAddress];
+        showTransactionStatus = true;
+        if (isLocalhost(endpointURI)) {
+            await submitCreateProvider(
+                localDotApi.api as ApiPromise,
+                undefined,
+                endpointURI,
+                signingKeys,
+                newProviderName,
+                addNewTxnStatus);
+        } else {
+            if (isFunction(web3FromSource) && isFunction(web3Enable)) {
+                const extensions = web3Enable('Frequency parachain provider dashboard: Creating provider');
+                if (extensions.length !== 0) {
+                    const injectedExtension = await web3FromSource(signingKeys.meta.source);
+                    await submitCreateProvider(
+                        localDotApi as ApiPromise,
+                        injectedExtension,
+                        endpointURI,
+                        signingKeys,
+                        newProviderName,
+                        addNewTxnStatus
+                    );
+                }
+            }
+        }
+    }
+
+    const addNewTxnStatus = (txnStatus: string) => {
+        txnStatuses = [...txnStatuses, txnStatus];
```

Comment:
I don't think all the txnStatuses are coming through as strings.

<img width="301" alt="image" src="https://github.com/LibertyDSNP/provider-dashboard/assets/1252199/c1d136f2-4f5f-4b26-bf5d-80b9e3e07261">

Reply:
Ah, right...this happens only on Mainnet right? I didn't want to keep testing this with mainnet because of the bug I found.  There's no way to make that extrinsic fail at the moment because it'll take anything as a provider name.

Reply:
This was on Testnet

Reply:
<TransactionStatus> component was not put into the RequestToBeProvider component.